### DISCUSSION TITLE
Debugging a nextjs build (Static worker unexpectedly exited with code: null and signal: SIGTERM)

### DISCUSSION THREAD

**Discussion by neilgd** [vercel/next.js#48192] [Help] [answered]
### Summary

Hi,

I have a nextjs app that runs fine in dev mode on my laptop, but when I try to deploy it I get the following error right at the end of the build:

`Static worker unexpectedly exited with code: null and signal: SIGTERM`

It seems to run successfully despite this error.

I was wondering if there was any way of troubleshooting this? I am not using eg getStaticProps anywhere, and I suspect it is a third party component that is causing this.

Thanks!

Neil

**Comment by sofiennenabli** [comment]
Hello im also having this issue any solution ?

**Comment by sofiennenabli** [comment]
I just killed the port used and restart the build and it works.

**Reply by neilgd** [reply]
That doesn't work for me unfortunately :(

**Reply by sofiennenabli** [reply, marked as answer]
did you try to delete .next folder and rebuild
rm -rf .next

**Reply by neilgd** [reply]
@sofiennenabli - thank you, that seems to have worked! Any idea why?

**Reply by scarf005** [reply]
this didn't work for us in CI on our private repo; our `.next` is `.gitignore`'d and the same error still happens

**Reply by hessnich** [reply]
@scarf005 did you solve this error, it seems you are more recent than most of the other posts? i am using bitbucket pipelines and trying to run `npm run build` in  a docker file. it builds locally, just not in the docker file

**Comment by MadaShindeInai** [comment]
#14 55.43 - info Generating static pages (21/29)
 
#14 55.78 Static worker unexpectedly exited with code: null and signal: SIGSEGV

Faced the same issue during deployment to railway / fly.io
Approved solution did not help

**Reply by fukemy** [reply]
any solution now sir?

**Comment by chukwumaokere** [comment]
Im getting this very error today on Vercel deployments.

**Comment by sinapirani** [comment]
i have this error
i deleted .next folder but still have this error

**Reply by Marx0xD** [reply]
hey, i do not know if this is still relevant but you should remove your node_modules and package-lock.json as well. then rebuild after npm install

**Reply by anuran-roy** [reply]
> hey, i do not know if this is still relevant but you should remove your node_modules and package-lock.json as well. then rebuild after npm install

This worked, thanks a lot!

**Reply by novertech411** [reply]
> hey, i do not know if this is still relevant but you should remove your node_modules and package-lock.json as well. then rebuild after npm install

this worked for me big time ...... delete the folder next ,node_modules, and package.lock.jason  ..... it worked big time

**Comment by ZentaChainAdmin** [comment]
**Update Dependencies:** 

Ensure that all your third-party dependencies are up-to-date. You can do this by going to your `package.json` file check and updating the version numbers of your dependencies. Once you've done this, run `npm install `or `yarn install` (depending on your package manager) to update your node_modules. In some cases, a dependency might be incompatible with others, so you might need to check the documentation of each dependency and see if they have any known issues with the versions you're using. You don't need to update next version if you are already using a stable version. Before Updating any dependencies make sure you check the documentation and update your code also with the version you are using. I recommend using node `v.18` and the build error on local or even on Vercel should be fixed.

In order to assist more efficiently conservation with the build issue it would be very helpful if people could share `package.json` file. to find out which dependencies gives the build error.

**Reply by emmabasseyokon** [reply]
<img width="1420" alt="Imagged" src="https://github.com/vercel/next.js/assets/109357895/44f62701-3b78-4b6e-ba10-ba3e17da84af">

**Comment by anttiriski** [comment]
Also had this problem today. Updated all the packages and now the build finished successfully.

I don't really know if that was the problem thought. I tried to redeploy a build from yesterday that was successful and today it gave the SIGTERM error. Weird stuff.

**Comment by vmarnauza** [comment]
Suddenly started having this error today for one of my deploy environments.

I was able to solve it by switching Node version to 16 in Project Settings. I then redeployed the app and it worked. Switched back to Node 18 and it still works.

**Comment by sinapirani** [comment]
I was able to identify the problem. The issue was located in the getInitialProps function in the _app.ts file. If there are any bugs present in that function, it will cause this error to be thrown.

**Reply by yunsii** [reply]
Encounter these warnings use App.getInitialProps function without any bugs. code: https://github.com/yunsii/nextjs-antd/blob/32addc413a15342504f8d0003d37c42e722e69ad/src/pages/_app.tsx#L17

**Comment by BhavyaCodes** [comment]
The error message disappeared for me after switching from node version 16 to 18.16.1

**Comment by chen-rn** [comment]
Running into this error as well w/ Next13 app directory, by build work fine locally but can't seem to work on Vercel.
I'm not using any getServerSideProps(directly at least)

I've tried:
- upgrading all packages
- node 16/18
- changing timeout duration to 1000

**Reply by chen-rn** [reply]
I realized that the error I was getting was SIGSEGV instead of SIGTERM. Anyhow, I couldn't get it to deploy on Vercel, while Railway.app worked.

I later made a completely new repo and just copy + pasted my code over and for some reason it worked. The only 2 library I didn't install this time were encoding + canvas, which is actually throwing an error in my vercel deployment.

Really not sure what the issue was... But at least it's working now?

**Reply by talb1985** [reply]
I have the problem when building inside a container image, it seems to fail on Collecting page data by printing the following error:

Next.js build worker exited with code: null and signal: SIGSEGV

**Comment by Allen-Wyatt** [comment]
I've had this error for a couple weeks now and none of the suggestions here have helped.  Any ideas?

**Comment by evanwinter** [comment]
I'm experiencing this too, deploying on Render, with Node `20.3.0` and Next.js `^13.4.10`. It's happened before in the last few weeks, but seems to just go away all by itself. Removing `.next` does not always fix it.

**Comment by Coderamrin** [comment]
I'm using next 13 with app dir 
just removing the .next and rebuilding it worked on local env

**Comment by a110605** [comment]
Same error after upgrade Next.js to `v13.5.2` and run yarn build under Node `v16.14.0`.
Removing .next folder does not help. Any suggestion to avoid / fix this error ? 

> `Static worker unexpectedly exited with code: null and signal: SIGINT`

**Comment by pbrissaud** [comment]
Same error after upgrade Next.js from `v13.4.19` to `v13.5.2` and run yarn build under Node `v18.17.1`

**Comment by kane50613** [comment]
same error happened building on vercel

**Comment by LeTamanoir** [comment]
I am also getting the same error when using `getInitialProps` at the `_app.tsx` level

**Comment by lzhgus** [comment]
Does anyone know the solution?

**Comment by evanwinter** [comment]
Does anyone have a minimal reproduction? I also have an _app.js level `getInitialProps` where there are intermittent errors so I'm wondering if that's it, but it sounds like that's not always the issue for some people.

**Comment by tomas-c** [comment]
There's an issue with a reproduction reported here:

https://github.com/vercel/next.js/issues/55817

**Comment by ADTC** [comment]
# Update:

Just updated to Next.js **13.5.6** with `npm update` and there's no more error.

Previous comments below:

# What didn't work for me:

First ensure that all version numbers have a caret before them in `package.json` (unless you have a specific requirement to fix to an exact version number, in which case I highly suggest adding a note somewhere explaining why).

Use `npm outdated` to check what's **red**. Use `npm update` to make the reds go away. Try deploying. If problem persists, use `npm outdated` to check what's _yellow_, manually update their version numbers in `package.json` and run `npm update` to make the yellows go away as well.

Proceed with caution though, especially with the yellows, because you are upgrading to a newer version which could potentially change the function of packages in a backwards-incompatible way. As always, test, test and test again.

# Where I'm getting the error:

I'm getting it in Vercel deployments, despite updating all packages with the above method. I have also tried re-deploying without build cache, to no avail.

It started after I updated Next.js version from 13.5.4 to 13.5.5:

```diff
-    "eslint-config-next": "^13.5.4",
+    "eslint-config-next": "^13.5.5",
-    "next": "^13.5.4",
+    "next": "^13.5.5",
```

Although the build is successful and deployed, the error still annoyingly appears in the log. It happens when static pages are generated.

```
TypeError [ERR_INVALID_STATE]: Invalid state: Controller is already closed                                                                      
    at new NodeError (node:internal/errors:405:5)                       
    at ReadableByteStreamController.enqueue (node:internal/webstreams/readablestream:1152:13)                                                   
    at c (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:27:1416)           
    at nJ (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:27:77012)         
    at Timeout._onTimeout (/vercel/path0/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.prod.js:27:77335)                                                                 
    at listOnTimeout (node:internal/timers:573:17)                      
    at process.processTimers (node:internal/timers:514:7) {             
  code: 'ERR_INVALID_STATE'                                             
 ✓ Generating static pages (13/13)                             
Static worker unexpectedly exited with code: 1 and signal: null
```

I can also reproduce this error locally by running `npm run build`. I have tried `rm -rf .next` to no avail. The error doesn't have any further detail about where in my code, there could be a mistake.

Since it started happening after an update of Next.js version and _**no changes**_ in my source code, I'm led to believe it's a new bug (or regression) in Next.js.

**Reply by ADTC** [reply]
I have narrowed the issue down to this change, and only this change:

```diff
-    "next": "^13.5.4",
+    "next": "^13.5.5",
```

This diff makes the following changes in the `package-lock.json` file:
```
"node_modules/next": "13.5.4" -> "13.5.5"

"dependencies":
"node_modules/@next/env": "13.5.4" -> "13.5.5"

"optionalDependencies":
"node_modules/@next/swc-darwin-arm64": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-darwin-x64": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-linux-arm64-gnu": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-linux-arm64-musl": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-linux-x64-gnu": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-linux-x64-musl": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-win32-arm64-msvc": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-win32-ia32-msvc": "13.5.4" -> "13.5.5"
"node_modules/@next/swc-win32-x64-msvc": "13.5.4" -> "13.5.5"
```

(There were _**no changes**_ in my source code.)

**Reply by ADTC** [reply]
I further narrowed it down to [this change with these commits](https://github.com/vercel/next.js/compare/v13.5.5-canary.0...v13.5.5-canary.1):

```diff
-    "next": "13.5.5-canary.0",
+    "next": "13.5.5-canary.1",
```

I'm not able to further narrow down because `npm intsall` refuses to work with commit IDs:

```json
"next": "vercel/next.js#e34fdf28900b3f0579adca8c7a73a150209f6283",
```

It gets stuck at `sill idealTree buildDeps` and I don't have a quick solution for that.

Edit: Okay, so it was apparently "getting stuck" because it was cloning the large repository which takes a while. You can see this if you run `npm install --verbose` and also watch the network traffic. So you just have to wait (a very long time) until the cloning finishes. **However, in the end it only has build errors, so I can't go on anyway.**

**Reply by ADTC** [reply]
Lols... just updated to Next.js **13.5.6** with `npm update` and guess what: no more error!

**Comment by sunech** [comment]
I experienced this issue as well, after upgrading from next 13.4.1 to 13.5.4.
Yesterday I tried upgrading to 13.5.5, which resolved it for me.

**Reply by ADTC** [reply]
Lol, what caused the issue for me, resolved it for you. That's just ridiculous; what's going on with Next.js? 😄

**Reply by sunech** [reply]
Yeah that's quite ironic :-) since it's such a generic error message, I would assume two different underlying causes to what you are experiencing and what I experienced.

**Reply by ADTC** [reply]
Lols... just updated to Next.js **13.5.6** with `npm update` and guess what: no more error!

**Reply by fukemy** [reply]
i got error at Next.js 14.2.5

**Comment by dancherb** [comment]
I've started receiving this error and experiencing crashes of our app on Heroku after updating one line of (completely unrelated and insignificant) Javascript code. When I apply a new commit to unchange this line of code, the error persists. However - when I roll back our deployment to a previous version, it no longer crashes. This suggests some kind of change (without consent) has happened under the hood either on the Next or Heroku side. Does anyone have some experience with this?

Update: looks like Heroku bumped the default Node version from v18 to v20, and this caused all future builds to crash. You can specify a specific Node version to use in your package.json, which is best practise.

**Comment by khteh** [comment]
`"next": "14.0.2",`

**Reply by arkhamRejek** [reply]
same on 
```
"next": "14.0.3"
```

**Comment by artrixdotdev** [comment]
getting this error on 

```
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Thu Oct 5 21:02:42 UTC 2023
Binaries:
  Node: 19.1.0
  npm: 8.19.3
  Yarn: 1.22.18
  pnpm: 8.10.2
Relevant Packages:
  next: 14.0.3
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
  
```

**Comment by stack-wuh** [comment]
<img width="971" alt="image" src="https://github.com/vercel/next.js/assets/34117238/8284ac42-96a5-40ac-85bf-5892456014f3">
what is error。。。

**Comment by nathanaelphilip** [comment]
Starting get this error a few days ago. Even after stripping out all but 2 generating slugs, it still can’t handle generating the data.

Local builds work fine, but Vercel seems to not be able to handle things.

**Comment by moniqx4** [comment]
I ran into the same error on build, using next14 and app router and it turned out to be because I had a page  in my app directory that in the page.tsx I had changed the function name from Page, to ContactPage ( app > contact > page.tsx ) , but I also had an component that was being imported to this page, also named ContactPage. Fixed by changing the function name in the app > contact > page.tsx to just be Contact.

**Comment by ilhamgum** [comment]
I encounter this error today on Vercel deployment build, been trying to redeploy 3 times, still got the same error, until I changed our NodeJS version on Vercel to 20.x (even though it's still beta), now we are happy as before:)

**Comment by JoKenPo** [comment]
I just update eslint-config-next to the right version and worked!
Was in version 13 while next was in version 14 
🤡

**Comment by in-ch** [comment]
I'm experiencing this too, with Node `20.9.0` and Next.js `^14.1.0`

**Comment by vanZeben** [comment]
Same as above. Node `20.11.0`, Next.js `^14.1.0`

Also worth noting, It seems this is a general error for build issues. That's totally fine but it should exit with an actual exit code so that build systems dont think the build was successful after its failed. What we're getting is silent failures here, which results in our production images rolling to a failed build because of this.

**Comment by diazemiliano** [comment]
Same as above. `Node 20.11.1`, `Next.js 14.1.3`.

**Comment by seivad** [comment]
Same issue as me, Node 18 or 20 with Next 14.1.x, works locally when building the docker containers on an M3 MBP but Github Actions throws this error during the build of Linux/Arm64 image for deployment to AWS EKS on graviton cpu's

**Comment by realamirhe** [comment]
I still have the issue, I only face the issue in ci/cd (note that there is no cache, hence there doesn't exist any `.next` directory)

**Reply by realamirhe** [reply]
next: 14.1.2
node: 18 (18.19-slim)

**Reply by gopal-chhetri** [reply]
how did you solve this issue. I am also facing this issue only during ci/cd in aws

**Comment by oliverphardman** [comment]
Using Next 14.1.4 and can't build on Amplify.

**Comment by julien-devatom** [comment]
I have the same issue in a Netlify environment: It's working well locally, in Github actions, but in Netlify it is throwing this error. 

And sometimes I just have this error: "Command failed with exit code 129"

**Comment by rxn7** [comment]
still happens.

**Reply by abdulkerimawad** [reply]
I have this error on Vercel and Netlify as well

![image](https://github.com/vercel/next.js/assets/79903897/e9113e86-499b-4653-8c99-6bdf6e867f2e)

**Reply by NutonC** [reply]
same issue i am facing. Is it solve now?

**Reply by cgen01** [reply]
Happening to me as well. Any updates on this?

**Comment by usmanmani1122** [comment]
Not sure if this helps but I made it work by tweaking the value of `CIRCLE_NODE_TOTAL`:
```
CIRCLE_NODE_TOTAL=8 yarn next build
```

**Comment by dbvisel** [comment]
This seems to have stopped for me on the new Next runtime on Netlify, though it was always intermittent for me there.

**Comment by sandilsranasinghe** [comment]
Does anybody know what causes this?

**Comment by randbytes** [comment]
Fixed this issue after fixing the `next/dynamic` component accessing null properties.
Tried to delete .next before with no success.
Node v21, next 14.1

**Comment by Redskinsjo** [comment]
same here, I _uploaded packages_, _removed .next_, _removed node_modules_ and _removed yarn.lock_, also passed _from node x20 to x18_ in Vercel, **nothing worked**.

It's not solved, but **only in a preview publish** (**Vercel**).

Merging the branch in main, and pushing main to production didn't encounter this error anymore.

**Comment by DOZBORNE** [comment]
I had this problem with my app. Version `^14.2.x` and node `20.12.2`. 

I could not figure out why it would build 26/27 of my static pages. However, on the last page to generate, it failed with the `Static worker unexpectedly exited with code: null and signal: SIGTERM` error. 

The fix ended up being updating our version of builder.io to version 3.2.6 from 3.1.0. There was an issue with declaring CSS inside builder, and then builder correctly building that page in next.js

**Comment by Redskinsjo** [comment]
found out that building on a Vercel preview for the first time leads to this error, if the server uri (the data point) where the requests hit is incorrect, or not yet existing.

**Comment by cgen01** [comment]
I was able to resolve this by adding this to my Next.JS config file:

```js
experimental: {
  workerThreads: false,
  cpus: 1,
},
```

**Reply by christian-draeger** [reply]
this worked. thx a lot

**Reply by fukemy** [reply]
not work to me

**Comment by Redskinsjo** [comment]
if you want to build pages, first push your models, schema, queries. Then push your pages in a second step. Because your instant build will certainly match the build in place (previous). Without your architecture in place, page generation fails. I just resolved it again.

**Comment by maipato10** [comment]
I have the same issue and none of the provided solutions work I get ⨯ Static worker exited with code: 130 and signal: null, npm run dev work fine but npm run start gives web socket connection fail , I am using trpc and other API calls other API call pass but trpc fails on npm run start

"engines": {
    "node": ">=20.0.0"
  },

 "next": "^14.2.4",

 "@prisma/client": "^4.14.1",
    "@tailwindcss/forms": "^0.5.3",
    "@tanstack/react-query": "^4.18.0",
    "@trpc/client": "^10.4.1",
    "@trpc/next": "^10.4.1",
    "@trpc/react-query": "^10.4.1",
    "@trpc/server": "^10.4.1",


  binaryTargets   = ["native", "darwin", "darwin-arm64", "debian-openssl-3.0.x"]

**Comment by SarthakSKumar** [comment]
Hi.

Error: `Static worker unexpectedly exited with code: null and signal: SIGTERM`,

Here's a solution that worked for me:

### Solution

1. **Delete the lock files**: This includes `package-lock.json` (if you're using npm) and `pnpm-lock.yaml` (if you're using pnpm).
2. **Remove the `node_modules` folder**: This will ensure that all your dependencies are reinstalled fresh.
3. **Delete the `.next` folder**: This will clear any cached builds.

### Steps to Resolve

1. **Remove lock files and folders**:
    ```sh
    rm -f package-lock.json pnpm-lock.yaml
    rm -rf node_modules .next
    ```

2. **Reinstall dependencies**:
    If you're using npm:
    ```sh
    npm install
    ```
    If you're using pnpm:
    ```sh
    pnpm install
    ```

3. **Rebuild your project**:
    ```sh
    npm run build
    ```
    or
    ```sh
    pnpm build
    ```

### Background

The issue occurred after making numerous changes across multiple files in the directory simultaneously. This approach I believe kinda resets the environment, ensuring there are no useless artefacts from previous builds or stale dependencies that could cause such issues.

### Environment

I was running **Next.js v.14.0.4** when I encountered and resolved this issue.

I hope this helps resolve the issue for y'all as well.

**Comment by chertik77** [comment]
still hapens, tried everything above

**Comment by justinherrera** [comment]
In my case, I fixed this error, when my database is from Google Cloud Platform and the error happens when the build is trying to fetch a data from my database but my deployment is not authorized to connect into it or it cannot reach the database server. I just went to GCP Connections and authorized my network / ip and it works now.

**Comment by ammarhairiHD** [comment]
I deleted and reinstalled .next, node_modules and package-lock.json, the result is still the same, 
Error fetching beta testers: Error: Failed to fetch beta testers or invalid content type
    at Object.c [as generateStaticParams] (/vercel/path0/young_guards/.next/server/app/(beta-tester)/beta-tester/[securepass]/page.js:1:4492)
    
  Try to deploy Next.js using Vercel, I would be glad if anyone can explain and fix the error.

**Comment by badmagick329** [comment]
started having this issue a few days ago. sometimes it just randomly works. then it stops working again. i've tried everything listed above

**Comment by fukemy** [comment]
found solution but not remember the the I solved this. Because `npm run build` seem run the project in production mode and check if any error => break build. All you need to do is make sure how to resolve all problem show in the terminal

**Comment by tarikwaleed** [comment]
the way i solved the problem is that i stopped the `pm2` process that runs the previous build

`pm2 stop <process_name>`

**Comment by krishmarsonia** [comment]
I did 2 things to resolve
1. delete the node_modules, package-lock.json and .next folder and files
2. I ran npm cache clean --force command

**Comment by eddundead** [comment]
Just try to go across your pages list and check for server side props (i.e. getServerSideProps, getStaticProps, getStaticPaths or similar calls for data retrieving). One of the page seems incorrect processing that or have no return in render or prepared props. Try to comment that pages one by one, after some tricky work you will have the success. That approach helped to me, one of my pages was built incorrect and wasn't even used. After removed my build has successfully run.

**Comment by TeemuKoivisto** [comment]
It probably just means your machine has run out of memory. I received first this one https://github.com/orgs/vercel/discussions/5360

If you're building on your own computer, try reinstalling and rebooting to purge your swap and all. For me that fixed it on M1 MBP. If you're on remote machine, probably the same or increase the memory.

**Reply by charleslcso** [reply]
> It probably just means your machine has run out of memory. I received first this one https://github.com/orgs/vercel/discussions/5360
> 
> If you're building on your own computer, try reinstalling and rebooting to purge your swap and all. For me that fixed it on M1 MBP. If you're on remote machine, probably the same or increase the memory.

Page gone...

**Comment by abhishekkovuritw** [comment]
Tried all the above things, but still facing the issue.

I have created this issue 
https://github.com/vercel/next.js/issues/74972

**Comment by RonAlmog** [comment]
This might sound stupid, but it worked for me, so i'll just put it here anyways: make sure you have a loading.tsx in the root folder.
in my case it's a search page that is using useSearchParams, and there is something with searchParam that needs to be inside a Suspense, and if it's not, then you get that dreaded error. not too friendly or helpful. Anyways, bottom line: if the page that is giving you the trouble is using searchParams, make sure you have that loading page.

**Reply by ben519** [reply]
Great find! This solved the issue for me. A few notes for the record..

1. (Before implementing this fix) `npm run build` only failed for me on the **second** build after deleting `.next` (i.e. the re-build).
2. The error I got was massive and cryptic, but this was at the tail end of it

    ```
    TypeError: Cannot read properties of undefined (reading 'length')
        at WasmHash._updateWithBuffer (/Users/bgorman/Projects/Satxuma/repos/satxuma-app/node_modules/next/dist/compiled/webpack/bundle5.js:28:1425886)
        at WasmHash.update (/Users/bgorman/Projects/Satxuma/repos/satxuma-app/node_modules/next/dist/compiled/webpack/bundle5.js:28:1425221)
        at BatchedHash.update (/Users/bgorman/Projects/Satxuma/repos/satxuma-app/node_modules/next/dist/compiled/webpack/bundle5.js:28:1421582)
        at /Users/bgorman/Projects/Satxuma/repos/satxuma-app/node_modules/next/dist/compiled/webpack/bundle5.js:28:302368
        at processQueue (/Users/bgorman/Projects/Satxuma/repos/satxuma-app/node_modules/next/dist/compiled/webpack/bundle5.js:28:1439460)
        at process.processTicksAndRejections (node:internal/process/task_queues:85:11)
    
    Node.js v22.13.1
    Static worker exited with code: 1 and signal: null
    ```

3. Adding `loading.tsx` to my app root solved the issue, but I still needed to delete my corrupted `.next` folder beforehand.

Here's a simple `loading.tsx` file that fixed the problem for me.

```typescript
// app/loading.tsx

export default function Loading() {
  return <div>Loading...</div>
}
```

**Reply by gurkankaya05** [reply]
Thanks for solution. It worked for me!!

**Comment by beemee-p** [comment]
in my case, fetching data is fetching during build time. this make failure to build. 
just add { cache: 'no-store' }  to fetch function and it worked!
it changes static to dynamic rendering ...

`fetch('/api/example', {cache: "no-store")`

**Comment by alex8023g** [comment]
Perhaps your server doesn't have enough memory or cores. Try to stop another apps on you're server while next building

**Reply by sathishkumar172003** [reply]
Yeah. I have stopped other pm2 processes and built it. It worked.

**Comment by badmagick329** [comment]
In case anyone else is having this issue due to lack of memory during build process and you're using docker, you can build the image on your machine and simply load it on the server. You can even output the image as a tar file if you don't want to push it to a registry. This solved it for me

**Comment by xuanthinhJB** [comment]
I follow the solution in this post and it works fine! [Nextjs export timeout configuration](https://stackoverflow.com/questions/70134643/nextjs-export-timeout-configuration)
Just increate the static page generation timeout:
```
// next.config.js
module.exports = {
  // time in seconds of no pages generating during static
  // generation before timing out
  staticPageGenerationTimeout: 1000,
}
```
